WooCommerce/Classes/ViewRelated/Editor/FormatBar/LinkSettings/LinkSettingsViewController.swift

```diff
@@ -0,0 +1,294 @@
+import UIKit
+import Yosemite
+
+struct LinkSettings {
+    var url: String
+    var text: String
+    var openInNewWindow: Bool
+    var isNewLink: Bool
+
+    init(url: String, text: String, openInNewWindow: Bool, isNewLink: Bool) {
+        self.url = url
+        self.text = text
+        self.openInNewWindow = openInNewWindow
+        self.isNewLink = isNewLink
+    }
+}
+
+enum LinkAction {
+    case insert
+    case update
+    case remove
+    case cancel
+}
+
+/// Allows the user to update the settings for a link text.
+///
+final class LinkSettingsViewController: UIViewController {
+    typealias LinkCallback = (_ action: LinkAction, _ settings: LinkSettings) -> ()
+
+    private var linkSettings: LinkSettings
+
+    /// Table Sections to be rendered
+    ///
+    private var sections = [Section]()
+    private let callback: LinkCallback?
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    init(linkSettings: LinkSettings, callback: @escaping LinkCallback) {
+        self.linkSettings = linkSettings
+        self.callback = callback
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigation()
+        configureMainView()
+        configureTableView()
+        configureSections()
+        registerTableViewCells()
+        updateNavigation()
+    }
+}
+
+// MARK: Actions
+//
+private extension LinkSettingsViewController {
+    func editURL() {
+        let placeholder = NSLocalizedString("Please enter a URL", comment: "Placeholder for editing the URL of a text link")
+        let navigationTitle = NSLocalizedString("URL", comment: "Navigation bar title for editing the URL of a text link")
+        let textViewController = TextViewViewController(text: linkSettings.url,
+                                                        placeholder: placeholder,
+                                                        navigationTitle: navigationTitle,
+                                                        keyboardType: .URL) { [weak self] text in
```

Comment:
What do you think about adding a capitalization property here? The TextViewViewController's textView could expose the `.autocapitalizationType` and be added as a property on init:

```
init ( // other properties here, text / placeholder / navigationTitle / etc.
.autocapitalizationType: UITextAutocapitalizationType = .sentences // default capitalization
)
```

Then the parameter to turn off capitalization for the URL textview would be:

`.autocapitalizationType = .none`

Reply:
sounds like a great idea! will update

Reply:
added here: https://github.com/woocommerce/woocommerce-ios/pull/1497/commits/f79f2f6be9f2798d959b30be5e20e3b2814693e2